### PR TITLE
Add circular slider

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

This circular slider will improve the round-slider with better mobile touch controls : 
- [x] drag control
- [x] tap control
- [x] touch friendly
- [x] keyboard support
- [x] single or dual mode

CleanShot 2023-06-20 at 19 13 21

CleanShot 2023-06-20 at 19 13 27

Demo : https://deploy-preview-16981--home-assistant-gallery.netlify.app/#components/ha-control-circular-slider

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-control-circular-slider.ts [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+    const highPercentage = this._valueToPercentage(highValue ?? this.max);
+
+    const lowArcLength = lowPercentage * f * maxRatio;
+    const lowStrokeDasharray = `${lowArcLength} ${f - lowArcLength}`;
+
+    const highArcLength = (1 - highPercentage) * f * maxRatio;
+    const highStrokeDasharray = `${highArcLength} ${f - highArcLength}`;
+    const highStrokeDashOffset = `${highArcLength + f * (1 - maxRatio)}`;
+
+    return svg`
```
Comment (bramkragten) [inline src/components/ha-control-circular-slider.ts:335]:
```suggestion
    return html`
```

`svg` is for when used inside a `svg` element

**Comment by frenck** [comment]
Didn't we decide that the little white circle showing the current value would be come a line instead?

**Comment by bramkragten** [comment]
keyboard support doesn't seem to work for me?

**Comment by piitaya** [comment]
Oh I see, it works when you focus the slider using keyboard, not mouse.
I will look to fix that.

File: src/components/ha-control-circular-slider.ts [hunk 2]
```diff
@@ -0,0 +1,546 @@
...
+    return lowDistance <= highDistance ? "low" : "high";
+  }
+
+  private _setActiveValue(value: number) {
+    const type = this._activeSlider ?? "value";
+    this[type] = value;
+  }
+
+  private _getActiveValue(): number | undefined {
+    const type = this._activeSlider ?? "value";
```
Comment (bramkragten) [inline src/components/ha-control-circular-slider.ts:186]: Same here

**Review by bramkragten — APPROVED** [review]

**Comment by milandzuris** [comment]
I want see this Slider type in Gauge and Thermostat @piitaya

**Comment by piitaya** [comment]
Something like this https://github.com/home-assistant/frontend/pull/17002 ? 👀

**Comment by milandzuris** [comment]
> Something like this #17002 ? 👀

yes, thx for your amazing work on frontend

**Pull request merged**